### PR TITLE
Update levels to optimize simulation results genetic.py #216, Update price_forecast.py  Calculate the electricity price from tibber

### PR DESCRIPTION
**Pull request by redmoon2711**
I took the idea from Celle and tested and adapted it. The levels can now be set via configuration:

self.state_levels = {
    'idle': (0, 4),
    'discharge': (5, 9),
    'ac': (10, 14),
    'dc': (15, 19)
}
The results with optimize_input_1.json are:
start_hour=0, ngen=300 after 3 iterations:

main:
Total balance_Euro 0.07506518757851482
Total losses 5518.667417355373

change:
Total balance_Euro -0.20943269628099148
Total losses 5671.084338842975

The goal is reached much faster with more states. ac_charge is no longer as dominant as before.

### CODE REVIEW

File: src/akkudoktoreos/prediction/price_forecast.py [hunk 1]
```diff
@@ -1,144 +0,0 @@
-import hashlib
```
Comment (Lasall) [inline src/akkudoktoreos/prediction/price_forecast.py:1]: to reset to a specific commit, try:
```
git reset --hard 213aa7b3d1a067980b3d4600e15e79205b02d873
```
or use revert. (You can find unreferenced commits with git reflog, they are available for 30 days. To use that commit on some branch, you can git cherry-pick xyz)

Comment (Lasall) [reply src/akkudoktoreos/prediction/price_forecast.py]: I didn´t say anything

Comment (redmoon2711) [reply src/akkudoktoreos/prediction/price_forecast.py]: Sorry, this is my first PR... I try my best :-)

Comment (redmoon2711) [reply src/akkudoktoreos/prediction/price_forecast.py]: I have replace the price_forcast.py, but i don't know how can i remove the price_forecast.py from the PR.

Comment (Lasall) [reply src/akkudoktoreos/prediction/price_forecast.py]: no worries, just reset to the Akkudoktor main branch and then put your commit on top and update this branch (force push):
```
# make sure to have Akkudoktor-EOS as remote "akkudoktor" in your git repo
git remote add akkudoktor https://github.com/Akkudoktor-EOS/EOS.git

# reset your current main branch to the akkudoktor github
git reset --hard akkudoktor/main

# Get your previous commit with the state changes (can look it up with git reflog) and push
git cherry-pick 340ad9da53ad3784f368bf8b95afaa0030e9057c
git push --force
```

For the other PR best create a feature branch (aka not the main branch, so you can keep that in sync with the akkudoktor repo)
```
# get latest changes of the akkudoktor github
git fetch akkudoktor

# create a new branch "prices"
git checkout -b prices akkudoktor/main

# Get your previous commit with the price calculation
git cherry-pick b886ef3c026a982423bed2093d6f42f072893fef

# Push to new branch (which is currently active, so it is also referenced as HEAD) on your fork
git push -u origin HEAD
```

**Comment by NormannK** [comment]
hi before committing the code you should run a "pre-commit" e.g. like this: `pre-commit run --files .\price_forecast.py` or for everything: `pre-commit run --all-files`
That way the code style is the same for the whole project and some easy mistakes are caught. 
Now its complaining about mypy stuff which I have not looked into yet. 
btw since you did not have a merged PR yet all checks here needs to be approved by a maintainer. Ill click the button as soon es I see it.

File: src/akkudoktoreos/prediction/price_forecast.py [hunk 2]
```diff
@@ -82,9 +83,28 @@ def is_cache_expired(self) -> bool:
...
+        - Taxes, levies, and surcharges: €5.214 per kWh
+        - Value-added tax (VAT): 19% (multiplied by 1.19)
+
+        Args:
+            euro_pro_mwh (float): The base electricity price in € per megawatt-hour (MWh).
+
+        Returns:
+            float: The final electricity price in € per kilowatt-hour (kWh).
+        """
+        return (euro_pro_mwh / 10 + 1.81 + 9.3 + 5.214) * 1.19
```
Comment (NormannK) [inline src/akkudoktoreos/prediction/price_forecast.py:106]: why are we dividing by 10? shouldn't it be 1000 to get from mWh to kWh?

**Comment by drbacke** [comment]
The state changes seem to be ok for me, but tibber price API is not ok, also the charges / spot price should be splitted. This is very important for direct marketing. Please remove that.

**Review by drbacke — CHANGES_REQUESTED** [review]
The state changes seem to be ok for me, but tibber price API is not ok, also the charges / spot price should be splitted. This is very important for direct marketing. Please remove that.

**Comment by drbacke** [comment]
Fixed the issues in https://github.com/Akkudoktor-EOS/EOS/pull/242
Sorry, but this is time critical and it was easier to fix it from Scratch. But thanks for the Help,😊

**Pull request closed without merging**